### PR TITLE
OBPIH-4914 Errors with loading / fetching redesigned list pages

### PR DESCRIPTION
**Pull request by kchelstowski**
I also fixed PO page, because when moving all the filter logic to `PurchaseOrderList`, the destination when changing location didn't change - it always stayed the same, because when we initially go to the PO page, we pass our current location as destination in the URL, but when changing location, the queryParams didn't change and this part:
```
 if (queryProps.destination) {
      defaultValues.destination = props.currentLocation.id === queryProps.destination
        ? currentLocationOption
        : await fetchLocationById(queryProps.destination);
    }
```
always took initially set destination from URl so it had to be changed to: 
`await fetchLocationById(props.currentLocation?.id);`

it was agreed with Katarzyna and me that I would fix it while doing this ticket, just wanted to explain it if you were wondering, why I changed this part also.

### CODE REVIEW

File: src/js/components/invoice/InvoiceListTable.jsx [hunk 1]
```diff
@@ -36,7 +38,22 @@ const InvoiceListTable = ({
 
   // Util ref for react-table to force the fetch of data
   const tableRef = useRef(null);
+
+  // Cancel token/signal for fetching data
+  const { CancelToken } = axios;
+  const sourceRef = useRef(CancelToken.source());
+
+  // eslint-disable-next-line consistent-return
```
Comment (drodzewicz) [inline src/js/components/invoice/InvoiceListTable.jsx:46]: remove this eslint warning and move the if condition inside return 
```
return () => {
    if (currentLocation?.id) sourceRef.current.cancel('Fetching canceled');
};
```

File: src/js/components/purchaseOrder/PurchaseOrderListTable.jsx [hunk 2]
```diff
@@ -57,11 +58,27 @@ const PurchaseOrderListTable = ({
 
   // Util ref for react-table to force the fetch of data
   const tableRef = useRef(null);
+
+  // Cancel token/signal for fetching data
+  const { CancelToken } = axios;
```
Comment (drodzewicz) [inline src/js/components/purchaseOrder/PurchaseOrderListTable.jsx:63]: you can probably just do `import { CancelToken } from 'axios';`

File: src/js/components/purchaseOrder/PurchaseOrderListTable.jsx [hunk 3]
```diff
@@ -57,11 +58,27 @@ const PurchaseOrderListTable = ({
...
+  const sourceRef = useRef(CancelToken.source());
+
   const fireFetchData = () => {
+    // Each time we fetch, we want to 'reset' the token/signal
+    sourceRef.current = CancelToken.source();
     tableRef.current.fireFetchData();
   };
   const isCentralPurchasingEnabled = supportedActivities.includes('ENABLE_CENTRAL_PURCHASING');
 
+  // eslint-disable-next-line consistent-return
```
Comment (drodzewicz) [inline src/js/components/purchaseOrder/PurchaseOrderListTable.jsx:73]: same as previouse, should probably move `if (...)` inside return

File: src/js/components/stock-movement/inbound/StockMovementInboundTable.jsx [hunk 4]
```diff
@@ -45,7 +46,23 @@ const StockMovementInboundTable = ({
 
   const tableRef = useRef(null);
 
-  const fireFetchData = () => tableRef.current.fireFetchData();
+  // Cancel token/signal for fetching data
+  const { CancelToken } = axios;
```
Comment (drodzewicz) [inline src/js/components/stock-movement/inbound/StockMovementInboundTable.jsx:50]: same import `CancelToken` straight from `axios`

**Review by drodzewicz — CHANGES_REQUESTED** [review]

**Review by drodzewicz — APPROVED** [review]
everything else looks good

File: grails-app/i18n/messages.properties [hunk 5]
```diff
@@ -2928,7 +2928,7 @@ react.invoice.buyerOrganization.label=Buyer organization
 react.invoice.createdBy.label=Created by
 react.invoice.viewDetails.label=View Invoice Details
 react.invoice.edit.label=Edit Invoice
-react.invoice.error.fetching.label=Could not fetch list of invoices
+react.invoice.error.fetching.label=Could not fetch list of invoices (most probably due to abort while changing location)
```
Comment (awalkowiak) [inline grails-app/i18n/messages.properties:2931]: Where did this label come from?

**Review by awalkowiak — COMMENTED** [review]
LGTM, but I strongly dislike the labels for the fetching errors or fails

File: grails-app/i18n/messages.properties (hunk 5 above)
Comment (kchelstowski) [reply grails-app/i18n/messages.properties:2931]: obviously from my imagination :( what would you suggest?

Comment (jmiranda) [reply grails-app/i18n/messages.properties:2931]: Why do we need to add more context here? I would just say "Unable to fetch invoices" and leave it at that.

Comment (jmiranda) [reply grails-app/i18n/messages.properties:2931]: Or "Unable to fetch invoices. Try refreshing the page."

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**